pkg/k8s/config/config.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"github.com/cilium/cilium/pkg/defaults"
+)
+
+// Configuration is the configuration interface for the k8s package
+type Configuration interface {
+	K8sAPIDiscoveryEnabled() bool
+}
+
+// DefaultConfiguration is an implementation of Configuration with default
+// values
+type DefaultConfiguration struct{}
+
+// NewDefaultConfiguration returns an implementation of Configuration with
+// default values
+func NewDefaultConfiguration() Configuration {
+	return &DefaultConfiguration{}
+}
+
+// K8sAPIDiscoveryEnabled returns true if API discovery of API groups and
+// resources is enabled
+func (d *DefaultConfiguration) K8sAPIDiscoveryEnabled() bool {
+	return defaults.K8sEnableAPIDiscovery
```

Comment:
Is this right? Shouldn't this be the value specified by the user?

Reply:
Yes, this represents the default configuration when no user options are available. This also clearly documents in which cases that happens instead of just relying on `pkg/option` which is uninitialized in several cases.